src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

```diff
@@ -709,8 +708,17 @@ public ValueTask<int> SendToAsync(ReadOnlyMemory<byte> buffer, SocketFlags socke
             saea.SetBuffer(MemoryMarshal.AsMemory(buffer));
             saea.SocketFlags = socketFlags;
             saea._socketAddress = socketAddress;
+            saea.RemoteEndPoint = null;
             saea.WrapExceptionsForNetworkStream = false;
-            return saea.SendToAsync(this, cancellationToken);
+            try
+            {
+                return saea.SendToAsync(this, cancellationToken);
+            }
+            finally
+            {
+                // detach user provided SA so we do not accidentally stomp on it later.
+                saea._socketAddress = null;
```

Comment:
It's ok to null it here because it's only used synchronously as part of the send operation and not asynchronously?

Reply:
yes. We will make  `_socketAddress` before first try. f we do not finish synchronously on Unix, it would be copied to queued `*SendOperation`. For Windows, the `const sockaddr *lpTo` is IN for `WSASendTo` and AFAIK it does not need to be preserved until the overlapped IO is completed.